### ISSUE TITLE
It is not possible to set accent color for the button inside the toolbar

### ISSUE THREAD

**Issue by atruschenkov** [angular/material#4928]
Hi,

Seems it is not possible to have buttons with any accent color within the toolbar. It works for icon button, but not for regular button. It is always defaults to white.

Here is the link to codepen:

https://codepen.io/anon/pen/JYbVrO

and the snippet:

```
<div ng-controller="AppCtrl" class="toolbardemoBasicUsage" ng-app="MyApp">
  <md-content>
    <md-toolbar>
      <div class="md-toolbar-tools">
        <md-button class="md-icon-button md-accent" aria-label="Settings">
          <md-icon md-svg-icon="img/icons/menu.svg"></md-icon>
        </md-button>
        <md-button class="md-accent" aria-label="Settings">
          I want accent color here!
        </md-button>
      </div>
    </md-toolbar>
  </md-content>
</div>
```

**Comment by pulse14** [comment]
:+1:

**Comment by EladBezalel** [comment]
Also `md-primary` is not working,

It seems to be in [L10](https://github.com/angular/material/blob/master/src/components/toolbar/toolbar-theme.scss#L10)
that compiles to:
``` css
md-toolbar:not(.md-menu-toolbar) .md-button {
    color: rgba(255,255,255,0.87);
}
```

But it might be intentionally

**Comment by pdore-netfore** [comment]
+1

**Comment by LionC** [comment]
> But it might be intentionally

I do not see a reason why it should, toolbars are used for top navigation which needs some way to highlight the current state - not allowing the accent color to be used for that would not make any sense.

It seems to be caused by [this commit](https://github.com/angular/material/commit/d9ba0e13a5abca05d6afa3b3b781b0dc7677183f) adding the `:not(.md-menu-toolbar)` selector to

    md-toolbar:not(.md-menu-toolbar) .md-button {
        color: rgba(255,255,255,0.87);
    }

thus making it more specific than

    .md-button.md-accent {
        color: rgb(255,64,129);
    } 

Removing the `:not`-selector fixes the problem. The best solution would likely be to have own rules for `.md-menu-toolbar` rather than using a `:not`-selector (which also makes more sense in my mind, as `md-menu-toolbar` is the exception rather than the rule isnt it?)

**Comment by LionC** [comment]
This was fixed for some time, but it is broken again since 1.0.6

working in 1.0.5: http://codepen.io/anon/pen/mPmWyJ
broken in 1.0.6: http://codepen.io/anon/pen/VabpYo

**Comment by fdev** [comment]
It's not just setting an accent color on a button within a toolbar, but also setting an accent color on the toolbar itself. The buttons (without `md-accent`) inside will retain the default color.

This variation on the issue was also fixed but broken again since 1.0.6.

working in 1.0.5: http://codepen.io/anon/pen/zqwMwd
broken in 1.0.6: http://codepen.io/anon/pen/xVdQdp

**Comment by ThomasBurleson** [comment]
This issue is closed as part of our ‘Surge Focus on Material 2' efforts.
      For details, see our forum posting @ http://bit.ly/1UhZyWs.
